packages/fuels-contract/src/script.rs

```diff
@@ -402,4 +463,344 @@ mod test {
+                .into_iter()
+                .map(|coin| {
+                    Input::coin_signed(
+                        fuel_tx::UtxoId::from(coin.utxo_id),
+                        coin.owner.into(),
+                        coin.amount.0,
+                        coin.asset_id.into(),
+                        0,
+                        0,
+                    )
+                })
+                .collect::<HashSet<_>>();
+
+            assert_eq!(expected_inputs, inputs_as_signed_coins);
+        }
+
+        #[test]
+        fn variable_outputs_appended_to_outputs() {
+            // given
+            let variable_outputs = [100, 200].map(|amount| {
+                Output::variable(random_bech32_addr().into(), amount, Default::default())
+            });
+
+            let calls = variable_outputs
+                .iter()
+                .cloned()
+                .map(|variable_output| {
+                    given_a_contract_call().with_variable_outputs(vec![variable_output])
+                })
+                .collect::<Vec<_>>();
+
+            // when
+            let (_, outputs) = Script::get_transaction_inputs_outputs(
+                &calls,
+                &random_bech32_addr(),
+                Default::default(),
+            );
+
+            // then
+            let actual_variable_outputs: HashSet<Output> = outputs[2..].iter().cloned().collect();
+            let expected_outputs: HashSet<Output> = variable_outputs.into();
+
+            assert_eq!(expected_outputs, actual_variable_outputs);
+        }
+    }
+
+    fn given_a_contract_call() -> ContractCall {
+        ContractCall {
+            contract_id: random_bech32_contract_id(),
+            encoded_args: Default::default(),
+            encoded_selector: [0; 8],
+            call_parameters: Default::default(),
+            compute_custom_input_offset: false,
+            variable_outputs: None,
+            external_contracts: Default::default(),
+            output_param: None,
+        }
+    }
+
+    impl ContractCall {
+        pub fn with_contract_id(self, contract_id: Bech32ContractId) -> Self {
+            ContractCall {
+                contract_id,
+                ..self
+            }
+        }
+        pub fn with_external_contracts(
+            self,
+            external_contracts: Vec<Bech32ContractId>,
+        ) -> ContractCall {
+            ContractCall {
+                external_contracts,
+                ..self
+            }
+        }
+
+        pub fn with_variable_outputs(self, variable_outputs: Vec<Output>) -> ContractCall {
+            ContractCall {
+                variable_outputs: Some(variable_outputs),
+                ..self
+            }
+        }
+
+        pub fn with_call_parameters(self, call_parameters: CallParameters) -> ContractCall {
+            ContractCall {
+                call_parameters,
+                ..self
+            }
+        }
+    }
```

Comment:
This should be inside the module defining `ContractCall` I would say

Reply:
These are extended only for testing purposes, they live only here. If we start needing these in production code we can move it into the impl of ContractCall inside ContractCall rs